frontend/apps/reader/modules/readerview.lua

```diff
@@ -487,6 +493,57 @@ function ReaderView:drawTempHighlight(bb, x, y)
 end
 
 function ReaderView:drawSavedHighlight(bb, x, y)
+    -- set/free note mark sign
+    if self.highlight.note_mark == "sidemark" then
+        if not self.note_mark_sign then
+            self.note_mark_sign = TextWidget:new{
+                text = "\u{F040}", -- pencil
+                face = Font:getFace("smallinfofont", 14),
+                padding = 0,
+            }
+        end
+    else
+        if self.note_mark_sign then
+            self.note_mark_sign:free()
+            self.note_mark_sign = nil
+        end
+    end
+    -- calculate position x of the note side line/mark
+    if self.highlight.note_mark == "sideline" or self.highlight.note_mark == "sidemark" then
+        local screen_w = Screen:getWidth()
+        local sign_w = self.highlight.note_mark == "sideline" and 3 or self.note_mark_sign:getWidth()
+        local sign_gap
+        if self.ui.paging then
+            sign_gap = Screen:scaleBySize(10) -- to screen edge (fixed position)
+            if BD.mirroredUILayout() then
+                self.note_mark_pos_x1 = sign_gap
+            else
+                self.note_mark_pos_x1 = screen_w - sign_gap - sign_w
+            end
+        else
+            sign_gap = Screen:scaleBySize(5) -- to text edge
+            local doc_margins = self.ui.document:getPageMargins()
+            local pos_x_r = screen_w - doc_margins["right"] + sign_gap -- mark in the right margin
+            local pos_x_l = doc_margins["left"] - sign_gap - sign_w -- mark in the left margin
+            if self.ui.document:getVisiblePageCount() == 1 then
+                if BD.mirroredUILayout() then
```

Comment:
(For the position in the left or in the right margin, I initially thought using the UI language direction was maybe not ideal, and the book language's RTL status (available in language.lua's `languages_rtl`), or page RTL paragraphs percentage would be better - but as we may not have any language code for PDF, and as crengine does not expose its RN_PAGE_MOSTLY_RTL flag to frontend, let's go with what you did :)